Refactoring to separate the logic of writing CXX code from the AST nodes corresponding to CXX code (#267)
val-lang/val: Sources/CodeGen/CXX/README.md

The transpiler will take Val AST and convert into a CXX AST corresponding to C++ code.
The C++ code is then generated by walking the CXX AST producing the textual representation for the Val module.

> The translation between Val AST and CXX AST is done in `CXXTranspiler` class.

Reviewer: ```suggestion
The translation between Val AST and CXX AST is done by `CXXTranspiler.
```

At this stage, we are looking at AST transformation and we completely ignore the format of the CXX code.

> The `CXXCodeWriter` class takes a CXX AST and produces actual CXX code, in a textual form.

Reviewer: ```suggestion
`CXXCodeWriter` takes a CXX AST and produces actual CXX code, in a textual form.
```
